app-text/texlive/texlive-2021.recipe

```diff
@@ -0,0 +1,958 @@
+# from texmfcnf.lua; therefore the two values should be kept in sync.
+# (As should everything else in texmf.cnf <-> texmfcnf.lua.)
+#
+###		TEXMFCACHE = $TEXMFSYSVAR;$TEXMFVAR
+#
+# Where generated fonts may be written.  This tree is used when the sources
+# were found in a system tree and either that tree wasn't writable, or the
+# varfonts feature was enabled in MT_FEATURES in mktex.cnf.
+#
+###		VARTEXFONTS = $TEXMFVAR/fonts
+#
+# On some systems, there will be a system tree which contains all the font
+# files that may be created as well as the formats.  For example
+#   TEXMFVAR = /var/lib/texmf
+# is used in many distros.  In this case, set VARTEXFONTS like this
+#VARTEXFONTS = $TEXMFVAR/fonts
+# and do not mention it in TEXMFDBS (but _do_ mention TEXMFVAR).
+#
+# Remove $VARTEXFONTS from TEXMFDBS if the VARTEXFONTS directory is below
+# one of the TEXMF directories (avoids overlapping ls-R files).
+fixKSearchPaths()
+{
+	TEXMFCNF=$sourceDir/texk/kpathsea/texmf.cnf
+	sed -i -e "s@TEXMFROOT =.*@TEXMFROOT = $dataDir/texlive@" $TEXMFCNF
+	sed -i -e "s@TEXMFDIST =.*@TEXMFDIST = $dataDir/texlive/texmf-dist@" $TEXMFCNF
+	sed -i -e "s@TEXMFMAIN =.*@TEXMFMAIN = $dataDir/texlive/texmf-dist@" $TEXMFCNF
+	sed -i -e "s@TEXMFLOCAL =.*@TEXMFLOCAL = /boot/system/non-packaged/data/texmf-dist@" $TEXMFCNF
+	sed -i -e "s@TEXMFSYSVAR =.*@TEXMFSYSVAR = $sharedStateDir/texlive@" $TEXMFCNF
+	sed -i -e "s@TEXMFSYSCONFIG =.*@TEXMFSYSCONFIG = $sysconfDir/texlive@" $TEXMFCNF
+	sed -i -e "s@TEXMFHOME =.*@TEXMFHOME = /boot/home/config/settings/texmf-dist@" $TEXMFCNF
+	sed -i -e "s@TEXMFVAR =.*@TEXMFVAR = /boot/home/config/var/texmf@" $TEXMFCNF
+	sed -i -e "s@TEXMFCONFIG =.*@TEXMFCONFIG = /boot/home/config/settings/texmf/config@" $TEXMFCNF
+	sed -i -e "s@VARTEXFONTS =.*@VARTEXFONTS = /boot/home/config/settings/texmf/fonts@" $TEXMFCNF
+	# rename current TEXMFCNF; easier than writing a more complex sed script!
+	sed -i -e "s@^TEXMFCNF@IGNOREDTEXMFCNF@" $TEXMFCNF
+	# redefine TEXMFCNF: {$SELFAUTOLOC,$SELFAUTODIR,$SELFAUTOPARENT}{,{/share,}/texmf{-local,}/web2c}
+	sed -i -e "s@%TEXMFCNF.*@TEXMFCNF={$dataDir/texlive/texmf-dist/web2c,/boot/system/non-packaged/data/texmf-dist/web2c,/boot/home/config/settings/texmf-dist/web2c}@" $TEXMFCNF
+}
+
+PATCH()
+{
+	# disable the install-bin-links target
+	sed -i -e 's/!WIN32/WIN32/g' am/bin_links.am
+
+	./reautoconf
+
+	# may need to run this super early to ensure everything gets the right value...
+	fixKSearchPaths
+}
+
+BUILD()
+{
+	mkdir -p ../build && cd ../build
+
+	if [ $targetArchitecture = x86_gcc2 ] || [ $targetArchitecture = x86 ]; then
+		HOST=i586-pc-haiku
+	elif [ $targetArchitecture = x86_64 ]; then
+		HOST=x86_64-unknown-haiku
+	fi
+
+	runConfigure --omit-buildspec --omit-dirs "binDir sbinDir" \
+		$sourceDir/configure \
+		--bindir=$prefix/bin --sbindir=$prefix/bin \
+		--disable-native-texlive-build \
+		--enable-shared --disable-static \
+		--with-banner-add="/Haiku" --with-editor='lpe +%d "%s"' \
```

Comment:
Why Pe? Can't we just use 
```
open +%d "%s"
```
here instead to use the sys-default editor?

Reply:
open is non-blocking and will not wait for the editor to exit. This is what lpe is for (for example this is used in git when waiting for the user to enter a commit message).

I don't know what this is used for, but maybe it is important to have this behavior.